lib/src/utils/screen_loading/screen_loading_manager.dart

```diff
@@ -0,0 +1,220 @@
+        startTimeInMicroseconds: startTimeInMicroseconds,
+      );
+      debugPrint('${APM.tag} starting screen loading trace — screenName: $screenName, startTimeInMicroseconds: $startTimeInMicroseconds');
+      _currentUiTrace?.didStartScreenLoading = true;
+      _currentScreenLoadingTrace = trace;
+      return true;
+    }
+    debugPrint('${APM.tag} failed to start screen loading trace — screenName: $screenName, startTimeInMicroseconds: $startTimeInMicroseconds');
+    debugPrint('${APM.tag} didStartScreenLoading: ${_currentUiTrace?.didStartScreenLoading}, isSameName: ${screenName ==
+        _currentUiTrace?.screenName}');
+    return false;
+  }
+
+  /// @nodoc
+  @internal
+  Future<bool> reportScreenLoading(
+    ScreenLoadingTrace? trace, {
+    required int endTimeInMicroseconds,
+  }) async {
+    int? duration;
+    final isScreenLoadingMonitoringEnabled =
+        await FlagsConfig.ScreenLoading.isEnabled();
+    if (!isScreenLoadingMonitoringEnabled) {
+      log('Unable to report Screen loading capture, as ${FlagsConfig.ScreenLoading.name} feature is disabled.',
+          name: APM.tag);
+    }
+
+    // Handles if [endScreenLoading] was called prematurely
+    // As the endScreenLoading only extends the current screen loading trace
+    if (prematurelyEndedTraces.contains(trace)) {
+      log(
+        'Ending screen loading prematurely, as Screen loading was ended before full capture.',
+        name: APM.tag,
+      );
+      duration = 0;
+    }
+
+    // Only report the first screen loading trace with the same name as the active UiTrace
+    if (trace?.screenName == _currentScreenLoadingTrace?.screenName &&
+        _currentUiTrace?.didReportScreenLoading == false) {
+      trace!.duration = (trace.endTimeInMicroseconds ?? 0) - (trace.startTimeInMicroseconds ?? 0);
+      _currentUiTrace?.didReportScreenLoading = true;
+
+      APM.reportScreenLoading(
+        trace.startTimeInMicroseconds,
+        duration ?? trace.duration!,
+        _currentUiTrace?.traceId ?? 0,
+      );
+      return true;
+    } else {
+      debugPrint(
+        '${APM.tag} failed to report screen loading trace — screenName: ${trace?.screenName}, '
+            'startTimeInMicroseconds: ${trace?.startTimeInMicroseconds}, '
+            'duration: $duration, '
+            'trace.duration: ${trace?.duration ?? 0}',
+      );
+      debugPrint(
+        '${APM.tag} didReportScreenLoading: ${_currentUiTrace?.didReportScreenLoading}, '
+            'isSameName: ${trace?.screenName == _currentScreenLoadingTrace?.screenName}',
+      );
+      _reportScreenLoadingDroppedError(trace!);
+    }
+    return false;
+  }
+
+  void _reportScreenLoadingDroppedError(ScreenLoadingTrace trace) {
+    final error = DropScreenLoadingError(
+      trace,
+    );
+    debugPrint('${APM.tag}: Droping the screen loading capture — $trace');
+    // Todo: !IMP - Check if this should only be logged
+    CrashReporting.reportHandledCrash(error, error.stackTrace);
```

Comment:
@abdelhamid-f-nasser Did we finalize the discussion on whether we really want to report a non-fatal if a screen loading trace has been dropped? This can clutter the user's dashboards with unnecessary non-fatals that they probably won't care about that much. A simple log would be helpful I think.

Reply:
It'll be both logged and reported on the client's dashboard as a non fatal crash 😅